sub-packages/bionemo-scdl/docs/scdl-schema.md

```diff
@@ -0,0 +1,140 @@
+# SCDL Schema
+
+Eric T. Dawson
+1 August 2025
+
+## Version
+
+0.0.9
+
+**Implementation Status:** ✅ Fully implemented and validated against this specification
+
+## Overview
+
+The SCDL schema defines the structure of a SCDL archive. This enables backwards compatibility,
+clear versions and updates, and robust, safe loading of SCDL archives to and from disk.
+
+## SCDL Archive Structure (v0.0.9)
+
+The SCDL archive is a directory containing a binary header file and a series of arrays.
+The header contains metadata about the file, such as the version, the endianness, and the arrays that are contained in the file.
+The arrays are stored in a contiguous block of memory and are *not* user-readable by design. Users should not
+have access to modify the header, which should only be modified by the SCDL library.
+
+### Archive Header
+
+The header is a binary file that contains the metadata for the archive. It is stored in the root of the archive.
+
+#### Header Fields
+
+- Magic Number: The magic number of the archive. This is stored as a 4 byte string. It is always 'SCDL'.
+
+- Version: The version of the SCDL schema. This is is stored as three 8-bit integers.
+
+  - Major version
+  - Minor version
+  - Point version
+
+- Endianness: The endianness of the archive. This is stored as a single integer based on an enum, but the value is always NETWORK (big endian).
+
+- Backend: The backend of the archive. This is stored as a single integer based on an enum.
+
+- Arrays: A list of arrays in the archive. This is stored as a list of arrays.
+
+  - Name: The name of the array. This is stored as a string.
+  - Length: The length of the array. This is stored as a single integer.
+  - Dtype: The dtype of the array. This is stored as a string based on an enum.
+  - (Optional) Shape: The shape of the array. This is stored as a list of integers.
```

Comment:
@yzhang123 try making this `[Optional]` instead of `(Optional)` to appease the merge queue formatter. 

If that doesn't work, as a nuclear fail safe, you can add this file [here](https://github.com/NVIDIA/bionemo-framework/blob/main/.mdformat.toml#L3).